pkg/activator/net/revision_backends.go

```diff
@@ -139,17 +140,30 @@ func newRevisionWatcher(ctx context.Context, rev types.NamespacedName, protocol
 	}
 }
 
-func (rw *revisionWatcher) probe(ctx context.Context, dest string) (bool, error) {
+// probe probes the destination and returns whether it is ready according to
+// the probe. If the failure is not compatible with having been caused by mesh
+// being enabled, notMesh will be true.
+func (rw *revisionWatcher) probe(ctx context.Context, dest string) (pass bool, notMesh bool, err error) {
 	httpDest := url.URL{
 		Scheme: "http",
 		Host:   dest,
 		Path:   network.ProbePath,
 	}
 
+	// We want to differentiate between 503 errors, which are compatible with
+	// failures being caused by the mesh being enabled, and generic probe
+	// failures. We don't want to fall back to ClusterIP for generic probe fails.
```

Comment:
Should we drop the specific 503 mentioning here maybe? It'd be tedious to go update that whenever we update `IsPotentialMeshErrorResponse`.

I guess we'd have to potentially adjust tests anyway too, so might not be as big a deal.

Reply:
yeah, but this comment is pretty wordy and overspecific either way. Updated.